corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py

```diff
@@ -82,7 +82,11 @@ def do_livequery(timing_context, restore_state, response, async_task=None):
                 sync_ids = discard_already_synced_cases(live_ids, restore_state)
         else:
             sync_ids = live_ids
+
+        dependent_ids = live_ids - set(owned_ids)
+        debug('updating synclog: live=%r dependent=%r', live_ids, dependent_ids)
         restore_state.current_sync_log.case_ids_on_phone = live_ids
+        restore_state.current_sync_log.dependent_case_ids_on_phone = dependent_ids
```

Comment:
Was this part of the bug? I don't see any other code in this PR that references this field. Trying to figure out if it was an oversight in the original writing of livequery?

Reply:
This was the bug. This field is referenced in the synclog model and is used when updating the synclog after a form submission. Without this field being populated all cases on the phone are assumed to be 'primary' cases and won't be purged from the synclog. 

https://github.com/dimagi/commcare-hq/blob/1992a5737194eef7c7546057b7fb08e4834d44be/corehq/ex-submodules/casexml/apps/phone/models.py#L636-L638

As you are aware, purging cases from the synclog is essential to the incremental sync algorithm since CC won't send cases to the phone that it thinks are already there.